lark/exceptions.py

```diff
@@ -116,44 +118,71 @@ def match_examples(self, parse_fn, examples, token_type_match_fallback=False, us
 
         return candidate[0]
 
+    def _format_terminals(self, terminals):
+        if self._terminals_by_name:
+            ts = []
+            for ter in terminals:
+                ts.append(self._terminals_by_name[ter].user_repr())
+        else:
+            ts = terminals
+        return "Expected one of: \n\t* %s\n" % '\n\t* '.join(ts)
+
+
 class UnexpectedEOF(ParseError, UnexpectedInput):
-    def __init__(self, expected, state=None):
+    def __init__(self, expected, state=None, _terminals_by_name=None):
```

Comment:
I don't think anywhere in Lark there are parameters that start with an underscore. Why have them here?

Reply:
To mirror the attribute name, but that isn't useful.